tray.c

```diff
@@ -182,13 +182,22 @@ CreatePopupMenus()
 
     hMenu = o.groups[0].menu; /* the first group menu is also the root menu */
 
+    /* Save menudata from submenu */
+    UINT_PTR menuData = minfo.dwMenuData;
+
     /* Set notify by position style for the top menu - gets automatically applied to sub-menus */
     minfo.fMask = MIM_STYLE;
     GetMenuInfo(hMenu, &minfo);
     minfo.dwStyle |= MNS_NOTIFYBYPOS;
     SetMenuInfo(hMenu, &minfo);
 
     if (o.num_configs == 1 && o.chead) {
+        /* Set main menu's menudata to first connection's submenu menudata */
+        minfo.fMask = MIM_MENUDATA;
+        GetMenuInfo(hMenu, &minfo);
+        minfo.dwMenuData = menuData;
```

Comment:
just use o.chead here. The result is the same and it is much more obvious what happens.